src/bsp/src/stm32/src/HostUart.cpp

```diff
@@ -62,11 +70,22 @@ bool HostUart::send(const uint8_t* buffer, uint16_t length) {
     return ret;
 }
 
-int32_t HostUart::receive(uint8_t* buffer, uint16_t length) const {
-    (void)buffer;
-    (void)length;
-    // TODO: Implement
-    return 0;
+bool HostUart::receive(uint8_t* buffer, uint16_t length) {
+    if (buffer == NULL || length > m_streamMemory.size()) {
+        m_logger.log(LogLevel::Warn, "Invalid parameters for UART receive");
+        return false;
+    }
+
+    m_receivingTaskHandle = xTaskGetCurrentTaskHandle();
+    while (CircularBuff_getLength(&m_stream) < length) {
+        // Gets notified everytime a new packet is appended to stream
+        ulTaskNotifyTake(pdTRUE, 500);
+    }
+    m_receivingTaskHandle = NULL;
```

Comment:
Why change the taskhandle everytime? in case the HostUart receives in different thread?

Reply:
Exactly